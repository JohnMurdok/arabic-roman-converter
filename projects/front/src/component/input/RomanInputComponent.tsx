import React, { useState } from 'react';
import { TextField } from '@mui/material';
import DEFAULT_INPUT_PROPS, { ROMAN_REG_EXP } from '../../constant/input';

// const DEBOUNCE_TIME = 2000;
const ROMAN_INPUT_PROPS = {
    ...DEFAULT_INPUT_PROPS,
    type: 'string',
    label: 'Roman number',
};

const RomanInputComponent = () => {
    const [error, setError] = useState(false);

    const onChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
        const { value } = ev.target;
        if (!value.match(ROMAN_REG_EXP)) {
            setError(true);
            return;
        }
        setError(false);
    };


    return (
        <TextField error={error} {...ROMAN_INPUT_PROPS} onChange={onChange} />
    );
};

export default RomanInputComponent;
