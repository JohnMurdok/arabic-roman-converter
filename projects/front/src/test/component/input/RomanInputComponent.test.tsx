import { render, screen } from '@testing-library/react';
import RomanInputComponent from '../../../component/input/RomanInputComponent';

describe('[COMPONENT] RomanInputComponent', () => {
    test('matches snapshot', async () => {
        render(<RomanInputComponent />);
        const textField = await screen.findByTestId('roman-input');
        expect(textField).toMatchSnapshot();
    });
});
