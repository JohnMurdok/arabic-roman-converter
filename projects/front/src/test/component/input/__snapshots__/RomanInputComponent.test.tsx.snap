// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] RomanInputComponent matches snapshot 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  data-testid="roman-input"
>
  <label
    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink="true"
    for=":r0:"
    id=":r0:-label"
  >
    Roman number
    <span
      aria-hidden="true"
      class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
    >
       
      *
    </span>
  </label>
  <div
    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
  >
    <input
      aria-invalid="false"
      class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      id=":r0:"
      required=""
      type="string"
      value="I"
    />
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-1in441m"
      >
        <span>
          Roman number
           
          *
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
